react-app/components/FileManager/hooks.js

```diff
@@ -0,0 +1,257 @@
+      };
+
+      files.forEach((file) => {
+        newFileMap[file.id] = {
+          ...file,
+          parentId: destination.id
+        };
+      });
+
+      return newFileMap;
+    });
+  }, []);
+
+  const createFolder = useCallback((folderName) => {
+    setFileMap((currentFileMap) => {
+      const newFileMap = { ...currentFileMap };
+      const parentId = currentFolderIdRef.current;
+      const currentFolderKeys = Object.keys(newFileMap);
+      const parentName = newFileMap[currentFolderIdRef.current].name || '';
+      const isDirectoryAdded = currentFolderKeys.some(
+        (key) => newFileMap[key].name === folderName && newFileMap[key].parentId === parentId
+      );
+
+      if (isDirectoryAdded) {
+        return newFileMap;
+      }
+
+      const newFolderId = uuidv4();
+
+      const newFolderContent = {
+        id: newFolderId,
+        name: folderName,
+        isDir: true,
+        parentId: currentFolderIdRef.current,
+        path: `/${parentName}`,
+        isNew: true,
+        isSaved: false,
+        childrenIds: [],
+        childrenCount: 0,
+        children: []
+      };
+
+      newFileMap[newFolderId] = newFolderContent;
+
+      const parent = newFileMap[currentFolderIdRef.current];
+      newFileMap[currentFolderIdRef.current] = {
+        ...parent,
+        childrenIds: [...parent.childrenIds, newFolderId],
+        children: [...parent.children, newFolderContent]
+      };
+
+      return newFileMap;
+    });
+  }, []);
+
+  const createFile = useCallback((fileName, file) => {
+    setFileMap((currentFileMap) => {
+      const newFileMap = { ...currentFileMap };
+      const parentName = newFileMap[currentFolderIdRef.current].name || '';
+      const currentFolderKeys = Object.keys(newFileMap);
+      const parentId = currentFolderIdRef.current;
+
+      const isFileAdded = currentFolderKeys.some(
+        (key) => newFileMap[key].name === fileName && newFileMap[key].parentId === parentId
+      );
+
+      if (isFileAdded) {
+        return newFileMap;
+      }
+
+      const newFileId = uuidv4();
+      const newFileContent = {
+        id: newFileId,
+        name: fileName,
+        isDir: false,
+        isSaved: false,
+        parentId: currentFolderIdRef.current,
+        path: `/${parentName}`,
+        isNew: true,
+        file
+      };
+
+      newFileMap[newFileId] = newFileContent;
+
+      const parent = newFileMap[currentFolderIdRef.current];
+
+      newFileMap[currentFolderIdRef.current] = {
+        ...parent,
+        childrenIds: [...parent.childrenIds, newFileId],
+        children: [...parent.children, newFileContent]
+      };
+
+      return newFileMap;
+    });
+  }, []);
+
+  return {
+    fileMap,
+    currentFolderId,
+    assetsKeyToDelete,
+    rootFolderIdFixed,
+    setCurrentFolderId,
+    resetFileMap,
+    deleteFiles,
+    moveFiles,
+    createFolder,
+    createFile
+  };
+};
+
+export const useFiles = (fileMap, currentFolderId) => {
```

Comment:
[curious] why is it called `use...`?

Reply:
Because it's a hook, in React hooks as good practice need to call them use as prefix